### PR TITLE
Move PriorBox to PriorBoxIE transformation from CommonOptimization pass

### PR DESCRIPTION
**Pull request by itikhono**
PriorBox to PriorBoxIE transformation was mistakenly part of CommonOptimization passes
ticket: 38900

### CODE REVIEW

**Review by ilyachur — APPROVED** [review]
Looks OK for me. Please check private plugins, maybe we also need to modify them.

File: inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp [hunk 1]
```diff
@@ -164,7 +166,9 @@ ie::ICNNNetwork::Ptr FrontEnd::convertNetwork(ie::ICNNNetwork& network) {
         ngraph::op::GenericIE::DisableReshape noReshape(nGraphFunc);
 
         ngraph::pass::Manager manager;
-
+        // WA: ConvertPriorBox must be executed before ConstantFolding pass
+        manager.register_pass<ngraph::pass::InitNodeInfo>();
+        manager.register_pass<ngraph::pass::ConvertPriorBox>();
```
Comment (gladilov-gleb) [inline inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp:171]: Should the same be done in myriad_plugin.cpp and HDDL plugin?

Comment (itikhono) [reply inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp:171]: As I can see, we don't call Opset1ToLegacy transformations in myriad_plugin.cpp and in HDDL plugin.

Comment (gladilov-gleb) [reply inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp:171]: At the same time, common optimization is called in both myriad_plugin.cpp and frontend.cpp. Do we need to move the call there?

Comment (itikhono) [reply inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp:171]: @gladilov-gleb ConvertPriorBox is not related to CommonOptimizations, it was a part CommonOptimization due to historical reasons. In an amicable way, this transformation should be included in Opset1ToLegacy, but due to the fact that the transformation must be called before 1st Constant folding, we can't move it that.

Comment (gladilov-gleb) [reply inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp:171]: > the transformation must be called before 1st Constant folding

Before frontend.cpp transformations in myriad_plugin.cpp are applied. Does that mean the even with your changes in frontend.cpp the first Constant Folding is happening before ConvertPriorBox transformation?

Comment (itikhono) [reply inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp:171]: It seems so.
Why is Common Optimization duplicated (frontend.cpp, myriad.cpp)?

Comment (gladilov-gleb) [reply inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp:171]: At current moment I'm not sure there is still reason for that, so all passes from myriad_plugin.cpp could be moved to frontend.cpp

Comment (gladilov-gleb) [reply inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp]: I merged the PR that resolves the issue for you https://github.com/openvinotoolkit/openvino/pull/2220

**Review by gladilov-gleb — APPROVED** [review]
LGTM

@Maxim-Doronin please start VPU validation for that

**Review by Maxim-Doronin — APPROVED** [review]

**Pull request merged**